src/main/java/persistence/sql/dml/QueryBuilder.java

```diff
@@ -0,0 +1,112 @@
+package persistence.sql.dml;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Table;
+import jakarta.persistence.Transient;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class QueryBuilder implements Database{
```

Comment:
쿼리 빌더는 데이터베이스에 SQL 쿼리를 수행하는 책임까지 한다면 많은 책임을 가지게 되는 것 같아요.
하나의 책임을 온전히 수행할 수 있도록 쿼리의 생성에 대한 로직만 구현해주시면 좋을 것 같아요 :)

```suggestion
public class QueryBuilder {
```